src/wayland-eglsurface.c

```diff
@@ -366,6 +364,8 @@ damage_thread(void *args)
                                                surface->ctx.damageThreadSync,
```

Comment:
I don't think we want to be holding the lock while we do this wait.

Reply:
Fixed, thanks